common/src/main/java/io/homeassistant/companion/android/common/data/MTLSHelper.kt

```diff
@@ -0,0 +1,145 @@
+        val keyFile = File(baseDir, "$filename.key")
+        val certFile = File(baseDir, "$filename.crt")
+
+        if (!keyFile.exists())return
+        if (!certFile.exists())return
+
+        var certificateInputStream: FileInputStream? = null
+        try {
+            val certificateFactory = CertificateFactory.getInstance("X.509")
+
+            val privateKeyContent = keyFile.readText()
+                .replace("-----BEGIN.*PRIVATE KEY-----".toRegex(), "")
+                .replace(System.lineSeparator().toRegex(), "")
+                .replace("-----END.*PRIVATE KEY-----".toRegex(), "")
+            val privateKeyAsBytes: ByteArray = Base64.decode(privateKeyContent, Base64.DEFAULT)
+            val keySpec = PKCS8EncodedKeySpec(privateKeyAsBytes)
+            val keyFactory: KeyFactory = KeyFactory.getInstance("RSA")
+            val privateKey = keyFactory.generatePrivate(keySpec)
+
+            // Get certificate
+            certificateInputStream = FileInputStream(certFile)
+            val certificate = certificateFactory.generateCertificate(certificateInputStream)
+
+            // Set up KeyStore
+            val keyStore: KeyStore = KeyStore.getInstance("AndroidKeyStore")
+            keyStore.load(null)
+            keyStore.setKeyEntry(
+                keyentryname,
+                privateKey,
+                null,
+                arrayOf(certificate)
+            )
+
+            // Remove files after importing them if tls_client.keep doesn't exists (useful for debugging purposes)
+            if (!File(baseDir, "$filename.keep").exists()) {
+                keyFile.delete()
+                certFile.delete()
+            }
+            importMsg = context.getString(R.string.mtls_cert_importmsg_message_ok)
+        } catch (ex: Exception) {
+            importErrorMsg = ex.localizedMessage
+        } finally {
+            certificateInputStream?.close()
+        }
+    }
+
+    fun init(context: Context) {
+        tryImportKeys(context)
+        loadKeyData()
+    }
+    private fun getMTLSKeyManagerForOKHTTP(): X509KeyManager {
+        val mtlsKeyManager = object : X509KeyManager {
+            override fun getClientAliases(
+                p0: String?,
+                p1: Array<out Principal>?
+            ): Array<String> {
+                return emptyArray()
+            }
+
+            override fun chooseClientAlias(
+                p0: Array<out String>?,
+                p1: Array<out Principal>?,
+                p2: Socket?
+            ): String {
+                return ""
+            }
+
+            override fun getServerAliases(
+                p0: String?,
+                p1: Array<out Principal>?
+            ): Array<String> {
+                return arrayOf()
+            }
+
+            override fun chooseServerAlias(
+                p0: String?,
+                p1: Array<out Principal>?,
+                p2: Socket?
+            ): String {
+                return ""
+            }
```

Comment:
If we aren't using these can we add some comments as to why?

Reply:
These needs to be implemented because X509KeyManager is Abstract; i'll add a relevant comment.
For our purposes these functions will never be called, these are only placeholders.

Reply:
Based on the [docs](https://developer.android.com/reference/javax/net/ssl/X509KeyManager) we should be using [X509ExtendedKeyManager](https://developer.android.com/reference/javax/net/ssl/X509ExtendedKeyManager) instead.